### PR TITLE
FAIR-3296: Add dictionary methods

### PR DESCRIPTION
**Pull request by RossBarnie**
- Includes changes from FAIR-3296-refactor which is purely a refactoring effort to remove duplication
- Adds various dictionary operations as scripts

### CODE REVIEW

File: common/constants.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import os
```
Comment (RossBarnie) [inline common/constants.py:1]: Generally add comments to explain constants

Comment (RossBarnie) [reply common/constants.py:1]: Also add description to readme

File: fair-api-datasets-create.py [hunk 2]
```diff
@@ -1,26 +1,35 @@
...
-                print(f'Created dataset: {data["code"]}')
-                print(f'View on the web at: {FAIR_URL}#/data/datasets/{data["code"]}')
-            else:
-                print(f'Expected 1 dataset in response - received {(data)}')
+if len(sys.argv) < 2:
+    print(f'Usage: {sys.argv[0]} <path to dataset definition json file> <--dry-run>')
+    exit(1)
 
-with open(definition_file()) as fh:
+with open(ensure_file_exists(sys.argv[1])) as fh:
```
Comment (RossBarnie) [inline fair-api-datasets-create.py:32]: Ambiguous name for function as not clear it returns the file.

**Review by rodrigobarnes — APPROVED** [review]
Reviewed with Ross; Happy with the changes. Some of the refactoring could be scaled back for clarity but works as is.

**Pull request merged**